sources_test/path.cpp

```diff
@@ -0,0 +1,163 @@
+
+    std::string path = util::Path::combine<std::string_view>(
+        false, "user", "Desktop", "kus.png");
+
+    ASSERT_STREQ(path.c_str(), "user/Desktop/kus.png");
+
+    std::string name = util::Path::getName(path, context_local);
+
+    EXPECT_STREQ(name.c_str(), "kus.png");
+}
+
+TEST(PathGetName, test_get_name_folder)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path =
+        util::Path::combine<std::string_view>(true, "user", "Desktop", "kus");
+
+    ASSERT_STREQ(path.c_str(), "user/Desktop/kus/");
+
+    std::string name = util::Path::getName(path, context_local);
+
+    EXPECT_STREQ(name.c_str(), "kus");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST(PathNormalizeFolderPath, test_normalize_folder_path_with_sep_postfix)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path =
+        util::Path::combine<std::string_view>(true, "user", "Desktop", "kus");
+
+    ASSERT_STREQ(path.c_str(), "user/Desktop/kus/");
+
+    std::string normalize_folder_path =
+        util::Path::normalizeFolderPath(path, true);
+
+    EXPECT_STREQ(normalize_folder_path.c_str(), "user/Desktop/kus/");
+}
+
+TEST(PathNormalizeFolderPath, test_normalize_folder_path_without_sep_postfix)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path =
+        util::Path::combine<std::string_view>(true, "user", "Desktop", "kus");
+
+    ASSERT_STREQ(path.c_str(), "user/Desktop/kus/");
+
+    std::string normalize_folder_path =
+        util::Path::normalizeFolderPath(path, false);
+
+    EXPECT_STREQ(normalize_folder_path.c_str(), "user/Desktop/kus");
+}
+
+TEST(PathNormalizeFolderPath, test_normalize_folder_path_with_sep_postfix_2)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path = "user/Desktop/kus";
+
+    std::string normalize_folder_path =
+        util::Path::normalizeFolderPath(path, true);
+
+    EXPECT_STREQ(normalize_folder_path.c_str(), "user/Desktop/kus/");
+}
+
+TEST(PathNormalizeFolderPath, test_normalize_folder_path_without_sep_postfix_2)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path = "user/Desktop/kus";
+
+    std::string normalize_folder_path =
+        util::Path::normalizeFolderPath(path, false);
+
+    EXPECT_STREQ(normalize_folder_path.c_str(), "user/Desktop/kus");
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST(PathGetRelativeToApp, test_get_relative_to_app_folder)
+{
+    DECLARE_LOCAL_CONTEXT;
+
+    std::string path = util::Path::combine<std::string_view>(
+        true, "fi??rst", "se con d", "thi!!rd");
+
+    ASSERT_STREQ(path.c_str(), "fi??rst/se con d/thi!!rd/");
+
+    std::string normalize_folder_path =
+        util::Path::normalizeFolderPath(path, true);
+
+    ASSERT_STREQ(normalize_folder_path.c_str(), "fi??rst/se con d/thi!!rd/");
+
+    std::string full_path =
+        "/home/arsen/Desktop/KusServer/fi??rst/se con d/thi!!rd/";
+
+    // /home/arsen/Desktop/KusServer/
```

Comment:
As for me, I don't have the `arsen` folder in my `home` directory. You need to acquirethe current path by standard means and compare the result with the result of kus-path module.